NBXplorer/Repository.cs

```diff
@@ -1181,10 +1223,20 @@ public async Task<TrackedTransaction[]> GetMatches(IList<NBitcoin.Transaction> t
 					transactionsPerScript.Add(output.ScriptPubKey, tx);
 				}
 			}
+			foreach(var kv in await GetOutPointToScript(outpoints))
+			{
+				if (kv.Value is null)
+					continue;
+				scripts.Add(kv.Value);
+				foreach (var tx in transactionsPerOutpoint[kv.Key])
+				{
+					transactionsPerScript.Add(kv.Value, tx);
+				}
+			}
 			if (scripts.Count == 0)
 				return Array.Empty<TrackedTransaction>();
 			var keyPathInformationsByTrackedTransaction = new MultiValueDictionary<TrackedTransaction, KeyPathInformation>();
-			var keyInformations = await GetKeyInformations(scripts.ToArray());
+			var keyInformations = await GetKeyInformations(scripts);
 			foreach (var keyInfoByScripts in keyInformations)
```

Comment:
I think we will also have to add the same for outpoints also

Reply:
Look above, we are adding the scripts of the inputs in the list.

Reply:
Just saw that, thats neat.

Reply:
I also think this is good.